library/src/scala/tasty/reflect/Printers.scala

```diff
@@ -637,21 +637,22 @@ trait Printers
 
           def keepDefinition(d: Definition): Boolean = {
             val flags = d.symbol.flags
-            def isCaseClassUnOverridableMethod: Boolean = {
+            def isUndecompilableCaseClassMethod: Boolean = {
               // Currently the compiler does not allow overriding some of the methods generated for case classes
               d.symbol.flags.is(Flags.Synthetic) &&
               (d match {
-                case DefDef("apply" | "unapply", _, _, _, _) if d.symbol.owner.flags.is(Flags.Object) => true
+                case DefDef("apply" | "unapply" | "writeReplace", _, _, _, _) if d.symbol.owner.flags.is(Flags.Object) => true
                 case DefDef(n, _, _, _, _) if d.symbol.owner.flags.is(Flags.Case) =>
                   n == "copy" ||
                   n.matches("copy\\$default\\$[1-9][0-9]*") || // default parameters for the copy method
                   n.matches("_[1-9][0-9]*") || // Getters from Product
+                  n == "writeReplace" || // Case class serialization
```

Comment:
I think the comment (and method name) is miss leading given that `writeReplace` is added on all objects, not just case objects.

Reply:
Actually, the previous case should be the one that handles the `writeReplace`. I will remove this line.